src/backend/storeManagers/legendary/commands/base.ts

```diff
@@ -0,0 +1,37 @@
+import { z } from 'zod'
+import path from 'path'
+import { hasGame } from '../library'
+
+export const LegendaryAppName = z
+  .string()
+  .refine((val) => hasGame(val), {
+    message: 'AppName was not found on account'
+  })
+  .brand('LegendaryAppName')
+export type LegendaryAppName = z.infer<typeof LegendaryAppName>
+
+export const LegendaryPlatform = z.enum(['Win32', 'Windows', 'Mac'] as const)
+export type LegendaryPlatform = z.infer<typeof LegendaryPlatform>
+
+export const NonEmptyString = z.string().min(1).brand('NonEmptyString')
+export type NonEmptyString = z.infer<typeof NonEmptyString>
+
+export const Path = z
+  .string()
+  .refine((val) => path.parse(val).root, 'Path is not valid')
+  .brand('Path')
```

Comment:
just out of curiosity, I understand this check happens at runtime, right?

what should we expect to see in heroic if a path is invalid or a string is empty? some message in the logs?

Reply:
This depends on how the data is validated. If `SomeSchema.parse(...)` is used, an exception is thrown if the validation does not succeed. `.safeParse` on the other hand will always return an object of type `{ success: true, data: T } | { success: false, error: ZodError }`
I chose to use exclusively `.parse` for now to be a little more aggressive in finding potential issues & because there isn't a clear error propagation path in most functions, if we end up using more Zod in the future, we could validate the data at the source instead (potentially with better error handling there)